vsintegration/src/FSharp.Editor/Options/EditorOptions.fs

```diff
@@ -94,15 +92,10 @@ type FormattingOptions =
     static member Default =
         { FormatOnPaste = false }
 
-[<Export>]
-[<Export(typeof<IPersistSettings>)>]
-type EditorOptions 
-    [<ImportingConstructor>] 
-    (
-        [<Import(typeof<SVsServiceProvider>)>] serviceProvider: IServiceProvider
-    ) =
-
-    let store = SettingsStore(serviceProvider)
+[<Shared; Export; Export(typeof<Settings.IPersistSettings>)>]
```

Comment:
I am not good at MEF, but from the dependency flow perspective - have you considered actually passing the store to EditorOptions instead of creating it inside?

Reply:
Yes, but I wanted to expose our codebase to MEF as little as possible 😆. We would probably have to export a factory, which would in turn create the store, depending on whether we're in VS host or not etc.

Reply:
The point is, the VS ISettingsManager service we end up using is not a MEF part anyway.